File: client-scalaz/src/main/scala/org/openapitools/client/api/WrappedResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WrappedResponse._

case class WrappedResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
`type`: Option[String],
item: Option[Any],
messages: Option[MessageListResponse],
returning: Option[String])

object WrappedResponse {
  import DateTimeCodecs._

  implicit val WrappedResponseCodecJson: CodecJson[WrappedResponse] = CodecJson.derive[WrappedResponse]
  implicit val WrappedResponseDecoder: EntityDecoder[WrappedResponse] = jsonOf[WrappedResponse]
  implicit val WrappedResponseEncoder: EntityEncoder[WrappedResponse] = jsonEncoderOf[WrappedResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/wrapped_response.clj
(ns sirqul-io-t-platform.specs.wrapped-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs. :refer :all]
            [sirqul-io-t-platform.specs.message-list-response :refer :all]
            )
  (:import (java.io File)))


(def wrapped-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :type) string?
   (ds/opt :item) any?
   (ds/opt :messages) message-list-response-spec
   (ds/opt :returning) string?
   })

(def wrapped-response-spec
  (ds/spec
    {:name ::wrapped-response
     :spec wrapped-response-data}))
